frontend/packages/metal3-plugin/src/components/dashboard/utils.ts

```diff
@@ -0,0 +1,63 @@
+import { StatusGroupMapper } from '@console/internal/components/dashboard/inventory-card/inventory-item';
+import { InventoryStatusGroup } from '@console/internal/components/dashboard/inventory-card/status-group';
+import { MachineKind, NodeKind } from '@console/internal/module/k8s';
+import { getMachineNode, getName } from '@console/shared';
+
+import { getHostStatus } from '../../utils/host-status';
+import { HOST_ERROR_STATES, HOST_PROGRESS_STATES, HOST_SUCCESS_STATES } from '../../constants';
+import { findNodeMaintenance, getHostMachine } from '../../selectors';
+import { getHostFilterStatus } from '../table-filters';
+
+const BMH_STATUS_GROUP_MAPPER = {
+  [InventoryStatusGroup.OK]: HOST_SUCCESS_STATES,
+  [InventoryStatusGroup.PROGRESS]: HOST_PROGRESS_STATES,
+  [InventoryStatusGroup.ERROR]: HOST_ERROR_STATES,
+};
+
+export const getBMHStatusGroups: StatusGroupMapper = (bmhs, { machines, nodes, maintenances }) => {
+  const groups = {
+    [InventoryStatusGroup.OK]: {
+      statusIDs: ['ready', 'provisioned'],
+      count: 0,
+      filterType: 'host-status',
+    },
+    [InventoryStatusGroup.ERROR]: {
+      statusIDs: ['error'],
+      count: 0,
+      filterType: 'host-status',
+    },
+    [InventoryStatusGroup.PROGRESS]: {
+      statusIDs: ['registering', 'provisioning'],
+      count: 0,
+      filterType: 'host-status',
+    },
+    [InventoryStatusGroup.NOT_MAPPED]: {
+      statusIDs: ['other'],
+      count: 0,
+      filterType: 'host-status',
+    },
+  };
+
+  bmhs.forEach((bmh) => {
+    const machine = getHostMachine(bmh, machines as MachineKind[]);
+    const node = getMachineNode(machine, nodes as NodeKind[]);
+    const nodeMaintenance = findNodeMaintenance(maintenances, getName(node));
+    const hostMultiStatus = getHostStatus({ host: bmh, nodeMaintenance });
```

Comment:
Nit: renaming `bmh` to host would align to what we're using in other places for baremetal host variable and simplify
```suggestion
    const hostMultiStatus = getHostStatus({ host, nodeMaintenance });
```